src/server/string_family.cc

```diff
@@ -572,14 +586,36 @@ MGetResponse OpMGet(fb2::BlockingCounter wait_bc, uint8_t fetch_mask, const Tran
       }
     }
 
-    auto it_res = db_slice.FindReadOnly(t->GetDbContext(), key, OBJ_STRING);
+    const bool found = expire_params.has_value() &&
+                       ExpireKey(shard, db_slice, expire_params.value(), key, op_args, ctx);
```

Comment:
this extra lookup can be avoided if a generic iterator type is stored in `DedupItem`